### PR TITLE
[app-metrics] Extract MetricParamsBuilder + typed device/network state

### PR DESCRIPTION
**Pull request by tsapeta**
Followup to #45414 addressing review feedback from @Ubax in https://github.com/expo/expo/pull/45414#discussion_r3193867220.

Stacked on top of #45423.

## Why

Before this PR, `DeviceConditions` (both platforms) returned `Map<String, Any>` with `expo.*` keys baked in, and `AppStartupMonitoring` / `AppStartupManager` separately wrote `expo.frameRate.*` keys at the call site. The `expo.*` namespace was scattered across four files. Hard to see "all the keys this metric emits" without grepping.

This refactor separates the OS-reading code from the wire-format conversion:

- **`DeviceConditions`** reads OS state into typed values (`DeviceState`, `NetworkState` on Android; `DeviceState` on iOS — `NetworkPath` was already typed). No string keys.
- **`MetricParamsBuilder`** is the single source of truth for all `expo.*` keys. Pure converter: takes typed inputs, produces the keyed map. Unit-testable without OS calls or Robolectric.
- **Call sites** (`AppStartupMonitoring.markInteractive`, `AppStartupManager.buildInteractiveParams`) shrink to a single `MetricParamsBuilder.build(...)` call.

## What

- New `DeviceState` struct + `ThermalState` enum (iOS, Swift `Sendable`).
- New `DeviceState` / `NetworkState` data classes + `ThermalState` / `NetworkTransport` enums (Android).
- New `MetricParamsBuilder` (one file per platform), takes `userParams + frameMetrics + deviceState + networkPath/State`, returns the `expo.*`-keyed map.
- Existing collision rule preserved — framework-emitted keys override user-supplied keys.
- 10 new pure-function tests on iOS, 11 on Android. No Robolectric needed; the converter takes data classes.

## Wire format

**Unchanged for the TTI path.** The `expo.*` keys and value types produced for `timeToInteractive` are byte-identical to what's emitted today, and the existing `AppStartupMonitoringTests` (iOS) still pass without modification.

There's one edge-case behavior change worth noting: when a future caller passes `networkPath: nil` to `MetricParamsBuilder.build`, iOS now omits the `expo.network.*` keys (matching Android), instead of emitting `connected: false, type: "unknown"`. The TTI emission always passes a non-nil path, so the live metric is unaffected — this is just an internal API consistency fix between the two platforms.

## Test plan

- [x] iOS tests: existing `AppStartupMonitoringTests` pass; new `MetricParamsBuilderTests` covers each input combination, including the nil-network edge case.
- [x] Android tests: `./gradlew :expo-app-metrics:testDebugUnitTest` passes (verified locally); new `MetricParamsBuilderTest` covers each input combination.
- [x] Manual: launch observe-tester on a device, trigger TTI, confirm the metric's `params` map has the same keys/values as before.

### CODE REVIEW

File: packages/expo-app-metrics/android/src/main/java/expo/modules/appmetrics/utils/DeviceConditions.kt [hunk 1]
```diff
@@ -9,89 +9,132 @@ import android.os.BatteryManager
...
+}
+
 /**
- * Snapshots of the device's environment (power, thermals, connectivity)
- * attached to metrics like `timeToInteractive` so regressions can be
- * correlated with conditions outside the app's control.
+ * A snapshot of the device's power, thermal, and battery state. Fields are
+ * nullable so missing OS data (no service, pre-API-29 thermal API, no
+ * battery sticky broadcast yet) can be expressed as `null` rather than a
+ * sentinel.
```
Comment (Ubax) [inline packages/expo-app-metrics/android/src/main/java/expo/modules/appmetrics/utils/DeviceConditions.kt]:
```suggestion
 * Current device power, thermal, and battery information.
 *
 * Fields may be `null` when Android does not provide the data,
 * such as on older OS versions or before battery info is available.
```

**Review by Ubax — COMMENTED** [review]
Let me know when the PR is ready for testing

**Review by Ubax — APPROVED** [review]
